src/client/pages/AssessmentSection/DataTable/Table/TableBody/TableBody.tsx

```diff
@@ -13,10 +30,115 @@ type Props = {
   data: TableData
 }
 
+type UseValidateProps = {
+  table: Table
+  rowsData: TypeRow[]
+}
+// ===================
+
+const useValidations = (props: UseValidateProps) => {
+  const { table, rowsData } = props
+  const assessmentOrig = useAssessment()
+  const cycle = useCycle()
+  const countryIso = useCountryIso()
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    const unsubscribe = dispatch(
+      addAppListener({
+        matcher: isAnyOf(
+          // AssessmentSectionActions.updateNodeValues.fulfilled,
+          AssessmentSectionActions.getTableData.fulfilled,
+          AssessmentSectionActions.setNodeValues
+        ),
+        effect: (_, { getState }) => {
+          const state = getState()
+          const { data } = state.pages.assessmentSection
+          const nodes: Array<NodeUpdate> = []
+
+          rowsData.forEach((row) => {
+            const { variableName } = row.props
+            const { cols } = row
+            cols.forEach((col) => {
+              const { colName } = col.props
+              if (!colName) return
+              const validateFns = col.props.validateFns?.[cycle.uuid] ?? row.props.validateFns?.[cycle.uuid]
+              if (!validateFns) return
+              const metaCache: AssessmentMetaCache = {
```

Comment:
this is created every time and for each row.
Maybe add this to a local state variable, that gets updated when table prop changes? what do you think?